src/ResourceManager/Compute.ManagedService/Commands.Compute.ManagedService/Tools/Move-AzureRmVhdVM.ps1

```diff
@@ -0,0 +1,237 @@
+            return;
+        }
+        else
+        {
+            $vhdFiles = Export-VMWithProgress $HyperVServer $HyperVVMName $ExportPath;
+        }
+
+        # Convert the various VHD/VHDX file formats to fixed VHD files
+        $convertedVhdPath = Join-Path $vhdxFolder "converted";
+        mkdir $convertedVhdPath -Force;
+
+        foreach ($vhdFile in $vhdFiles)
+        {
+            $vhdFileName = Split-Path -Leaf -Path $vhdFile;
+            $destFile = Join-Path $convertedVhdPath $vhdFileName.Replace('.vhdx', '.vhd');
+            Write-Host "Converting '$($vhdFile)' to '$destFile'...";
+            if (Test-Path $destFile)
+            {
+                Write-Warning "Path already exists '$destFile'; please delete & retry...";
+            }
+            else
+            {
+                Convert-VHD -Path $vhdFile -DestinationPath $destFile;
+            }
+            $diskFiles += $destFile;
+        }
+    }
+
+    if ($diskFiles.Count -ge 1)
+    {
+        $rgName = $ResourceGroupName;
+        if ([string]::IsNullOrEmpty($rgName))
+        {
+            $rgName = ($HyperVVMName + (date).Ticks).Substring(0, 21);
+        }
+        New-AzureRmVhdVM -ResourceGroupName $rgName -Location $Location -OSType $OSType -DiskFile $diskFiles;
+    }
+    else
+    {
+        Write-Warning "Cannot find the VHD files for VM '$HyperVVMName'. Exit.";
+    }
+}
+
+function Get-VMOperatingSystemName($computerNameInput, $hyperVVMNameInput)
+{
+    # Filter for parsing XML data
+    filter Import-CimXml
+    {
+        # Create new XML object from input
+        $CimXml = [Xml]$_;
+        $CimObj = New-Object -TypeName System.Object;
+
+        # Iterate over the data and pull out just the value name and data for each entry
+        foreach ($CimProperty in $CimXml.SelectNodes("/INSTANCE/PROPERTY[@NAME='Name']"))
+        {
+            $CimObj | Add-Member -MemberType NoteProperty -Name $CimProperty.NAME -Value $CimProperty.VALUE;
+        }
+
+        foreach ($CimProperty in $CimXml.SelectNodes("/INSTANCE/PROPERTY[@NAME='Data']")) 
+        {
+            $CimObj | Add-Member -MemberType NoteProperty -Name $CimProperty.NAME -Value $CimProperty.VALUE;
+        }
+        # Display output
+        $CimObj;
+    }
+
+    # Get the virtual machine object
+    $query = "Select * From Msvm_ComputerSystem Where ElementName='" + $hyperVVMNameInput + "'";
+    $Vm = gwmi -Namespace root\virtualization\v2 -Query $query -ComputerName $computerNameInput;
+
+    # Get the KVP Object
+    $query = "Associators of {$Vm} Where AssocClass=Msvm_SystemDevice ResultClass=Msvm_KvpExchangeComponent";
+    $Kvp = gwmi -Namespace root\virtualization\v2 -Query $query -ComputerName $computerNameInput;
+
+    #Write-Host "Guest KVP information for: '$hyperVVMNameInput'";
+
+    # Filter the results
+    try
+    {
+        $osNameItem = $Kvp.GuestIntrinsicExchangeItems | Import-CimXml | where Name -eq "OSName";
+        Write-Output $osNameItem.Data;
```

Comment:
You should only write to the output if this is the end result of the cmdlet - if this is informational, it should go into the Progress, Verbose, or Debug streams

Reply:
If it is necessary information , the consider how it should be represented in the final cmdlet output

Reply:
removed.